network/transport/protocol.go

```diff
@@ -36,4 +36,20 @@ type Protocol interface {
 	PeerDiagnostics() map[PeerID]Diagnostics
 	// Version returns the version of the protocol
 	Version() int
+	// MethodName returns the fully qualified name of the gRPC stream.
+	MethodName() string
+}
+
+type Connection interface {
+	// Send tries to send the given message over the stream of the given protocol.
+	// If there's no active stream for the protocol, or something else goes wrong, an error is returned.
+	// A sender may specify ignoreSoftLimit=true to allow extra messages to be sent.
+	// This is needed to finish sending a TransactionList that falls within a single page.
+	Send(protocol Protocol, envelope interface{}, ignoreSoftLimit bool) error
+
+	// Peer returns the associated peer information of this connection.
+	Peer() Peer
+
+	// ID returns the internal identifier of the connection. This is a unique value through
```

Comment:
code comment not complete

Reply:
removed. was just a shortcut to Peer().ID